src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/WaterWayCheck.java

```diff
@@ -0,0 +1,526 @@
+        line.getAtlas().lineItemsContaining(line.asPolyLine().last(), this.waterwayTagFilter::test)
+                .forEach(waterways::add);
+
+        waterways.removeIf(line::equals);
+        final Location last = line.asPolyLine().last();
+        return waterways.stream().anyMatch(tLine -> tLine.asPolyLine().contains(last)
+                && !last.equals(tLine.asPolyLine().last()));
+    }
+
+    /**
+     * Check if a location is to the right of a line (e.g., kerbs, cliffs, and oceans)
+     *
+     * @param line
+     *            The line to compare the location to
+     * @param location
+     *            The location
+     * @return {@code true} if the location is to the right of the line. If the location is on the
+     *         line, or to the left of the line, we return {@code false}.
+     */
+    public boolean isRightOf(final PolyLine line, final Location location)
+    {
+        final PolyLine tLine = new PolyLine(location);
+        final Segment closest = line.segments().stream()
+                .min(Comparator.comparingDouble(s -> s.shortestDistanceTo(tLine).asMeters()))
+                .orElse(null);
+        if (closest != null)
+        {
+            final PolyLine testLine = new PolyLine(closest.first(), closest.last(), location);
+            final Angle difference = testLine.headingDifference().orElse(null);
+            if (difference != null && difference.asDegrees() > 0)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check if a location is inside an atlas bounds
+     *
+     * @param atlas
+     *            The atlas
+     * @param location
+     *            The location to check
+     * @return {@code true} if the atlas contains the location inside its bounds
+     */
+    public boolean isValidEndToCheck(final Atlas atlas, final Location location)
+    {
+        return atlas.bounds().fullyGeometricallyEncloses(location);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return !isFlagged(object.getOsmIdentifier()) && object instanceof LineItem
+                && this.waterwayTagFilter.test(object);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final LineItem line = (LineItem) object;
+        final Location last = line.asPolyLine().last();
+        final Location first = line.asPolyLine().first();
+        final Atlas atlas = line.getAtlas();
+        CheckFlag flag = null;
+        if (line.isClosed())
+        {
+            flag = createFlag(object,
+                    this.getLocalizedInstruction(FALLBACK_INSTRUCTIONS.indexOf(CIRCULAR_WATERWAY),
+                            object.getOsmIdentifier()),
+                    Collections.singletonList(line.asPolyLine().first()));
+        }
+        final double incline = this.elevationUtils.getIncline(first, last);
```

Comment:
I elected to not do this for two reasons:
1) The elevation data may not be "fine" enough to determine elevation between two nearby points (and won't be, in most cases).
2) There may be a ditch that is lower than the surrounding terrain that is not accounted for in the elevation data (highly likely with SRTM data).

Both (1) and (2) would be ameliorated if the elevation data is sufficiently fine. However, public domain data from the SRTM NASA mission have ~30m resolution, at best. Other data sources may be in the public domain and may have sufficient resolution for this, but I was more focused on general incline for the line instead of specific incline between points. I suppose we could check that the incline between each point in the line is downwards, and then have a specific instruction set for that.

Reply:
That's fair! To avoid over complicating the elevation part of the check I think it's fine as is. If we ever come across a dataset with finer resolution we could pull this out into a new check to look at the problem more closely

Reply:
I've added a comment to indicate that it can be expanded in the future, if it makes sense with available data sets.